cmd/controller/app/controller.go

```diff
@@ -100,33 +122,33 @@ func Run(opts *options.ControllerOptions, stopCh <-chan struct{}) {
 				continue
 			}
 
-			wg.Add(1)
 			iface, err := fn(ctx)
 			if err != nil {
 				log.Error(err, "error starting controller")
 				os.Exit(1)
 			}
-			go func(n string, fn controller.Interface) {
-				defer wg.Done()
+
+			g.Go(func() error {
 				log.V(logf.InfoLevel).Info("starting controller")
 
 				workers := 5
-				err := fn.Run(workers, stopCh)
-
-				if err != nil {
-					log.Error(err, "error starting controller")
-					os.Exit(1)
-				}
-			}(n, iface)
+				return iface.Run(workers, rootCtx.Done())
+			})
 		}
 
 		log.V(logf.DebugLevel).Info("starting shared informer factories")
-		ctx.SharedInformerFactory.Start(stopCh)
-		ctx.KubeSharedInformerFactory.Start(stopCh)
-		ctx.GWShared.Start(stopCh)
-		wg.Wait()
+		// TODO: we should wait for these informers to finish
```

Comment:
I can't think of a way to wait for the informers to finish, do we really need to have a `TODO` here? Could we instead have a comment that says that we have no way to wait for the informers goroutines to stop?

Reply:
I don't think there's a tonne of value in waiting for them to finish personally (given these are top-level `cmd/` packages anyway)

Reply:
Ok, I just removed that comment